fortitude/src/rules/correctness/error_handling.rs

```diff
@@ -0,0 +1,94 @@
+use std::iter::once;
+
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{AstRule, FromAstNode};
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_source_file::SourceFile;
+use tree_sitter::Node;
+
+/// ## What does it do?
+/// If the status of an `allocate` statement is checked by passing a variable to
+/// the `stat` argument, that variable must be checked.
+///
+/// ## Why is this bad?
+/// By default, `allocate` statements will crash the program if the allocation
+/// fails. This is often the desired behaviour, but to provide for cases in
+/// which the user wants to handle allocation errors gracefully, they may
+/// optionally check the status of an `allocate` statement by passing a variable
+/// to the `stat` argument:
+///
+/// ```f90
+/// allocate (x(BIG_INT), stat=status)
+/// if (status /= 0) then
+///   call handle_error(status)
+/// end if
+/// ```
+///
+/// However, if the `stat` variable is not checked, the program will continue to
+/// run despite the allocation failure, which can lead to undefined behaviour.
```

Comment:
Without wanting this to become a textbook, maybe we should mention `errmsg` here too?

Reply:
What do you think we should say about it? Should we just highlight that it's probably a good idea to set both?

Reply:
Do you think a rule to enforce always setting both would be a good idea?

Reply:
Yes, makes sense as a separate rule